File: OLSource1/v1/VS_csharp/codesnippet/VisualBasic/how-to--retrieve-data-from-a-query-by-using-code_1.vb
        Private Sub Good_Customer_Discount(ByVal entity As Order_Detail)
            For Each cust As Customer In _
                Me.DataWorkspace.NorthwindData.TopNSalesOrders(10)
                If cust.CustomerID = entity.Order.Customer.CustomerID Then
                    entity.Discount = 0.1
                End If
            Next

        End Sub

File: OLSource1/v1/vs140/codesnippet/CSharp/how-to--retrieve-data-from-a-query-by-using-code_1.cs
        private void Good_Customer_Discount(Order_Detail entity)
        {
            foreach (Customer cust in this.DataWorkspace.NorthwindData.
         TopNSalesOrders(10))
            {
                if (cust.CustomerID == entity.Order.Customer.CustomerID)
                {
                    entity.Discount = 0.1F;
                }
            }
        }